ngraph/core/reference/include/ngraph/runtime/reference/experimental_detectron_topk_rois.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <ngraph/runtime/host_tensor.hpp>
+#include <vector>
+#include "ngraph/node.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "ngraph/ops.hpp"
+#include "ngraph/shape_util.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            template <typename T>
+            void experimental_detectron_topk_rois(const T* input_rois,
+                                                  const T* input_probs,
+                                                  const Shape& input_rois_shape,
+                                                  const Shape& input_probs_shape,
+                                                  size_t max_rois,
+                                                  T* output_rois)
+            {
+                const size_t input_rois_num = input_rois_shape[0];
+                const size_t top_rois_num = std::min(max_rois, input_rois_num);
+
+                std::vector<int64_t> idx(input_rois_num);
+                std::iota(idx.begin(), idx.end(), int64_t{0});
+                std::sort(idx.begin(), idx.end(), [&input_probs](int64_t i1, int64_t i2) {
```

Comment:
What if the size of `input_probs` less than `input_rois_num`?

Reply:
It is impossible, because, according to the lines 68-75 of the file `ngraph/core/src/op/experimental_detectron_topkrois.cpp` (in the infer function of the operation `ExperimentalDetectronTopKROIs`), we must have `input_rois_shape[0] == rois_probs_shape[0]`, and, according to lines 61-67, the `input_probs` tensor must be 1D tensor.